### PR TITLE
[devel] Resign Leadership

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

*(Can you describe what functional change your PR is trying to effect?)*

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### CODE REVIEW

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/5140/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/5191/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/5212/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/5215/

File: arangod/Agency/ResignLeadership.cpp [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+
+  // Minimum 1 DB server must remain:
+  if (availServers.size() == 1) {
+    LOG_TOPIC("deaa0", ERR, Logger::SUPERVISION)
+        << "DB server " << _server << " is the last standing db server.";
+    return false;
+  }
+
+  // Check that each shard has an in sync follower
+  // TO BE IMPLEMENTED
```
Comment (neunhoef) [inline arangod/Agency/ResignLeadership.cpp]: No, this is intentional. The `resignLeadership` job should always be fast, it cannot help a customer who has chosen replicationFactor 1. We should remove the "TO BE IMPLEMENTED" and say so in a comment here.

File: js/actions/api-cluster.js [hunk 2]
```diff
@@ -869,7 +870,7 @@ actions.defineHttp({
 
     if (!req.isAdminUser) {
       actions.resultError(req, res, actions.HTTP_FORBIDDEN, 0,
-        'only allowed for admins on the _system database');
+        'only allowed for admins users');
```
Comment (neunhoef) [inline js/actions/api-cluster.js]: typo "admin"

**Pull request merged**